lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -39,26 +39,93 @@ def object
               object
             end
           end
+
+          ResourceIdentifier = Struct.new(:id, :type) do
```

Comment:
It representes a [resource identifier object](http://jsonapi.org/format/#document-resource-identifier-objects). But yeah, agreed, should be in the class header.